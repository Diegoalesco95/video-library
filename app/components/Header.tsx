'use client';

// @packages
import { useContext } from 'react';
import Link from 'next/link';
import Image from 'next/image';
import clsx from 'clsx';

// @scripts
import { AuthContext } from 'app/context/auth.context';
import useLogout from 'app/hooks/useLogout';
import gravatar from 'app/utils/gravatar';

// @styles
import styles from 'app/styles/layout/header.module.scss';

const Header = () => {
	const {
		state: { isLoggedIn, user },
	} = useContext(AuthContext);
	const { onLogout } = useLogout();

	const handleLogout = () => {
		onLogout();
	};

	return (
		<header className={clsx(styles.header)}>
			<Link href='/'>
				<span className={styles.header__logo}>Video Library</span>
			</Link>
			<div className={styles.header__menu}>
				<div className={styles['header__menu--profile']}>
					{isLoggedIn && user ? (
						<>
							<Image className={styles['header__menu--gravatar']} src={gravatar(user.email)} width={50} height={50} alt={user.email} />
							<p>{user.name}</p>
						</>
					) : (
						<>
							<Image src='/img/profile-2.png' width={50} height={50} alt='user Icon' />
							<p>Menu</p>
						</>
					)}
				</div>
				<ul>
					{isLoggedIn ? (
						<li>
							<Link href='/account'>Account</Link>
						</li>
					) : null}
					{isLoggedIn ? (
						<li>
							<Link href='/login' onClick={handleLogout}>
								Logout
							</Link>
						</li>
					) : (
						<>
							<li>
								<Link href='/login'>Login</Link>
							</li>
							<li>
								<Link href='/signup'>Sign up</Link>
							</li>
						</>
					)}
				</ul>
			</div>
		</header>
	);
};

export default Header;
