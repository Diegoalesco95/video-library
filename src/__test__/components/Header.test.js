import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';

describe('Header component', () => {
  test('Match Snapshot', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
