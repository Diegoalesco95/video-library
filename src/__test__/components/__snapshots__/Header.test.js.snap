// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Match Snapshot 1`] = `
<header
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Logo Platzi Video"
      className="header__img"
      src="test-file.stub"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt="user Icon"
        src="test-file.stub"
      />
      <p>
        Menu
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesi√≥n
        </a>
      </li>
      <li>
        <a
          href="/signUp"
          onClick={[Function]}
        >
          Registrarse
        </a>
      </li>
    </ul>
  </div>
</header>
`;
